### DISCUSSION TITLE
How do you feel about an installation script for qtile?

### DISCUSSION THREAD

**Discussion by SweepingBishops** [qtile/qtile#3497] [Ideas]
There seems to be a lot of people having trouble installing qtile from source. An installation script should potentially solve the problem.
Thoughts?

**Comment by elParaguayo** [comment]
I'd prefer to keep the docs up to date. It really should be a case of making sure all dependencies are installed before trying to build qtile.

If you're doing an installation script, how are you installing non-python packages? Different distros have different package managers and trying to cater for multiple distros in one script sounds messy.

My preference is therefore *not* to have an installation script but welcome other people's thoughts.

**Reply by m-col** [reply]
I guess an installation script can take the form of a snippet in the docs for a given distro.

So the Ubuntu installation page (it seems to be ubuntu and debian that have the most trouble) can have a few lines of code showing how to install.

How does that sound?

**Reply by elParaguayo** [reply]
Yeah. It feels like we need better instructions on distro specific installations.

Also feels like I'm going to need to set up a VM to test some of this...

**Reply by m-col** [reply]
Ha I was just thinking the same thing. Leave it to me, @elParaguayo, you seem to be working on many other things atm!

**Reply by SweepingBishops** [reply]
I actually did an install of qtile  (stable, xorg)  in a fresh VM of Ubuntu 20.04. I was able to do it in just two commands.
'pip install xcffib' and 'pip install qtile'
Create a desktop file, and done.

**Reply by m-col** [reply]
A fresh 20.04 VM has all the necessary X dependencies already then I guess? Which would make sense.

What about pip itself?

And indeed... python3?

**Reply by SweepingBishops** [reply]
Oh yeah had to install pip too. 
Python 3.8 is installed by default.

**Reply by elParaguayo** [reply]
> Ha I was just thinking the same thing. Leave it to me, @elParaguayo, you seem to be working on many other things atm!

Ha! I am losing track of what I'm supposed to be doing at the moment!

**Reply by SweepingBishops** [reply]
I started going through the installation page of the docs today, but it is going to take a lot of time to finish.

**Reply by 3nprob** [reply]
A properly packaged `.deb` release should be possible to make installable without additional manual steps (perhaps with a `pip install` step if there is some particular python requirement making this annoying)  and should serve the same role as an installation script (any such script will necessarily be distro-specific either way)

Seems ultimately more straighttforward and less maintenance effort to use standard tools available rather than reinventing the wheel

**Comment by clotodex** [comment]
FYI @SweepingBishops https://github.com/clotodex/qtile-install-script I have not tested it for X11, but I am using this for every source update of qtile. And now also for local development (it is way easier to just run the install instead of hoping that the locally patched version plays well with linked libraries and FFI stuff).
You just configure it in a config.toml (you can configure it to automatically install extensions like qtile-extras too)
Then run `./qtile-manager.py install`
